dev/com.ibm.ws.ejbcontainer.singleton.startup_fat/fat/src/com/ibm/ws/ejbcontainer/singleton/startup/fat/tests/PropertyBlockWorkDisabledTest.java

```diff
@@ -0,0 +1,101 @@
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package com.ibm.ws.ejbcontainer.singleton.startup.fat.tests;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.EnterpriseArchive;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.websphere.simplicity.ShrinkHelper.DeployOptions;
+
+import componenttest.annotation.Server;
+import componenttest.annotation.TestServlet;
+import componenttest.annotation.TestServlets;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+import singleton.property.web.AllowWorkNoStartupServlet;
+import singleton.property.web.AllowWorkStartupServlet;
+
+/**
+ * Test the behavior of the com.ibm.websphere.ejbcontainer.blockWorkUntilAppStartedWaitTime
+ * property for applications with and without Startup Singleton beans when the
+ * com.ibm.websphere.ejbcontainer.blockWorkUntilAppStarted property has been disabled.
+ *
+ * During application startup of an application that contains Startup Singleton bean,
+ * outside work will be blocked. The default time that this work will be blocked before
+ * an exception is thrown in 2 minutes. However, if the "blockWorkUnitlAppStartedWaitTime"
+ * property is set then outside work will be blocked until the application has completed
+ * startup processing or until the "blockWorkUntilAppStartedWaitTime" threshold has been met.
+ *
+ * During application startup of an application that does not contain a Startup Singleton bean,
+ * outside work will not be blocked.
+ *
+ * The following property values have been set in the TestBuild.xml:
+ * -Dcom.ibm.websphere.ejbcontainer.blockWorkUntilAppStartedWaitTime=20
+ * -Dcom.ibm.websphere.ejbcontainer.blockWorkUntilAppStarted=false
+ */
+@RunWith(FATRunner.class)
+public class PropertyBlockWorkDisabledTest extends FATServletClient {
+
+    @Server("PropertyBlockWorkDisabledServer")
+    @TestServlets({ @TestServlet(servlet = AllowWorkNoStartupServlet.class, contextRoot = "SingletonPropertyTest"),
+                    @TestServlet(servlet = AllowWorkStartupServlet.class, contextRoot = "SingletonPropertyTest") })
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+
+        // Use ShrinkHelper to build the ears
+
+        // -------------- SingletonPropertyIntf ------------
+        JavaArchive SingletonPropertyIntf = ShrinkHelper.buildJavaArchive("SingletonPropertyIntf.jar", "singleton.property.shared.");
```

Comment:
```suggestion
        JavaArchive SingletonPropertyIntf = ShrinkHelper.buildJavaArchive("SingletonPropertyIntf.jar", "singleton.property.shared");
```

Clarification: What is the purpose of the trailing `.`?  I see other buckets use `.*` to include sub-packages recursively. I believe in this case the Shrinkwrap implementation is handling the trailing `.` when it does package URL scanning. To avoid confusion, I'd recommend removing the trailing `.` or include the wildcard `.*`

This comment is valid for the rest of this method and for the PropertyBlockWorkDefaultTest